notification/src/handlers/storageHandler.ts

```diff
@@ -38,7 +63,7 @@ export class StorageHandler implements NotificationRepository {
      *
      * @returns     a Promise, containing either a Connection on success or null on failure
      */
-    public async initConnection(retries: number, backoff: number): Promise<void> {
+    private async initConnection(retries: number, backoff: number): Promise<Connection|null> {
```

Comment:
Wouldn"t expose the connection to outside of class. If something goes wrong, reject promise instead of returning null

Reply:
I think the connection is always exposed to the application.
You can always get the connection from the library via **getConnection()** function.

![image](https://user-images.githubusercontent.com/56654666/85198496-76e40300-b2e9-11ea-8212-ce858d9aa024.png)

Reply:
The issue with exposing the Connection is that we cannot wrap it with an implementation-neutral interface... Maybe we want to add another implementation with another library / database in the future.. That's why we shouldn't expose those implementation details ;-)